packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -50,14 +51,19 @@ function updateColumnsWidth(columns: GridColumns, viewportWidth: number) {
   }
 
   let newColumns = columns;
-  if (viewportWidth > 0 && numberOfFluidColumns) {
+  if (numberOfFluidColumns) {
     const flexMultiplier = viewportWidth / flexDivider;
-    newColumns = columns.map((column) => {
-      return {
-        ...column,
-        width: column.flex! ? Math.floor(flexMultiplier * column.flex!) : column.width,
-      };
-    });
+    newColumns = columns.map((column) =>
+      viewportWidth > 0
+        ? {
+            ...column,
+            width: column.flex! ? Math.floor(flexMultiplier * column.flex!) : column.width,
```

Comment:
I think you can do it in 1 line as flexMultiplier would be 0 if viewport is 0 and we merge col with the default type and string is default.
`width: column.flex! && flexMultiplier ? Math.floor(flexMultiplier * column.flex!) : column.width,`

Reply:
> I think you can do it in 1 line as flexMultiplier would be 0 if viewport is 0 and we merge col with the default type and string is default.
> `width: column.flex! && flexMultiplier ? Math.floor(flexMultiplier * column.flex!) : column.width,`

The `column.width` might not be the default one. The updated widths are saved to the state and passed as argument.

https://github.com/mui-org/material-ui-x/blob/c992eb3de48ed3f465cb8d3daacd45f13758224a/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts#L280

---

What about if we do the cloning only when the column is fluid?

```diff
diff --git a/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts b/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts
index 82559053..5525a787 100644
--- a/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts
+++ b/packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts
@@ -53,17 +53,18 @@ function updateColumnsWidth(columns: GridColumns, viewportWidth: number) {
   let newColumns = columns;
   if (numberOfFluidColumns) {
     const flexMultiplier = viewportWidth / flexDivider;
-    newColumns = columns.map((column) =>
-      viewportWidth > 0
-        ? {
-            ...column,
-            width: column.flex! ? Math.floor(flexMultiplier * column.flex!) : column.width,
-          }
-        : {
-            ...column,
-            width: column.flex! ? GRID_STRING_COL_DEF.width : column.width,
-          },
-    );
+    newColumns = columns.map((column) => {
+      if (!column.flex) {
+        return column;
+      }
+      return {
+        ...column,
+        width:
+          viewportWidth > 0
+            ? Math.floor(flexMultiplier * column.flex!)
+            : GRID_STRING_COL_DEF.width,
+      };
+    });
   }
   return newColumns;
 }
```

Reply:
> What about if we do the cloning only when the column is fluid?

That will work, I'll update it.

Reply:
@m4theushw done